toolkit/tools/scheduler/schedulerutils/printresults.go

```diff
@@ -33,6 +34,99 @@ func PrintBuildResult(res *BuildResult) {
+
+	graphMutex.RLock()
+	defer graphMutex.RUnlock()
+
+	failedSRPMs := make(map[string]*pkggraph.PkgNode)
+	failures := buildState.BuildFailures()
+	for _, failure := range failures {
+		failedSRPMs[failure.Node.SrpmPath] = failure.Node
+	}
+
+	prebuiltSRPMs := make(map[string]*pkggraph.PkgNode)
+	builtSRPMs := make(map[string]*pkggraph.PkgNode)
+	unbuiltSRPMs := make(map[string]*pkggraph.PkgNode)
+	unresolvedDependencies := make(map[string]bool)
+
+    buildNodes := pkgGraph.AllBuildNodes()
+    for _, node := range buildNodes {
+        if buildState.IsNodeCached(node) {
+            prebuiltSRPMs[node.SrpmPath] = node
+            continue
+        } else if buildState.IsNodeAvailable(node) {
+            builtSRPMs[node.SrpmPath] = node
+            continue
+        }
+
+		_, found := failedSRPMs[node.SrpmPath]
+		if !found {
+            unbuiltSRPMs[node.SrpmPath] = node
+        }
+    }
+
+	for _, node := range pkgGraph.AllRunNodes() {
+		if node.State == pkggraph.StateUnresolved {
+			unresolvedDependencies[node.VersionedPkg.String()] = true
+		}
+	}
+
+    csv_string := "Package,State,Blocker"
+
+	for srpm := range builtSRPMs {
+        csv_string += "\n" + filepath.Base(builtSRPMs[srpm].SrpmPath) + ",Built,,,"
+	}
+
+	for srpm := range prebuiltSRPMs {
+        csv_string += "\n" + filepath.Base(prebuiltSRPMs[srpm].SrpmPath) + ",PreBuilt,,,"
+	}
+
+	for srpm := range failedSRPMs {
+        node := failedSRPMs[srpm]
+        csv_string += "\n" + filepath.Base(node.SrpmPath) + ",Failed,"
+
+        // Failed nodes shouldn't have any blockers
+        blocking_nodes_str := ""
+	    fromNodes := pkgGraph.From(node.ID())
+	    for fromNodes.Next() {
+		    fromNode := fromNodes.Node().(*pkggraph.PkgNode)
+            if _, found := failedSRPMs[fromNode.SrpmPath]; found {
+                blocking_nodes_str += filepath.Base(fromNode.SrpmPath) + "-FAIL "
+            }
+            if _, found := unbuiltSRPMs[fromNode.SrpmPath]; found {
+                blocking_nodes_str += filepath.Base(fromNode.SrpmPath) + "-UNBUILT "
+            }
+        }
+
+        csv_string += blocking_nodes_str
+	}
+
+	for srpm := range unbuiltSRPMs {
+        node := unbuiltSRPMs[srpm]
+        csv_string += "\n" + filepath.Base(node.SrpmPath) + ",Unbuilt,"
+
+        blocking_nodes_str := ""
+	    fromNodes := pkgGraph.From(node.ID())
+	    for fromNodes.Next() {
+		    fromNode := fromNodes.Node().(*pkggraph.PkgNode)
+            if _, found := failedSRPMs[fromNode.SrpmPath]; found {
+                blocking_nodes_str += filepath.Base(fromNode.SrpmPath) + "-FAIL "
+            }
+            if _, found := unbuiltSRPMs[fromNode.SrpmPath]; found {
+                blocking_nodes_str += filepath.Base(fromNode.SrpmPath) + "-UNBUILT "
+            }
+        }
+
+        csv_string += blocking_nodes_str
+	}
+
+    f, _ := os.Create(outputPath)
```

Comment:
I think we should handle the `error` just incase we are unable to create the file for some reason.
The `WriteString` can `panic`, if the file was not created. and we do not have a sane handle to `f`